### ISSUE TITLE
[Qt6] GSSAPI path not correct when using vcpkg

### ISSUE THREAD

**Issue by ylz-at** [microsoft/vcpkg#23782]
**Describe the bug**
When I don't add `CMAKE_TOOLCHAIN_FILE`to the CMake command,
CMake finds GSSAPI in `GSS.framework`
```
Found GSSAPI: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/GSS.framework
```

If I use vcpkg by configuring `CMAKE_TOOLCHAIN_FILE`,
CMake finds GSSAPI in `usr/lib/libgssapi_krb5.tbd` which is a symlink,
```
Found GSSAPI: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/lib/libgssapi_krb5.tbd
```

And this leads to linker error,
```
Undefined symbols for architecture arm64:
  "___gss_c_nt_hostbased_service_oid_desc", referenced from:
      qGSsapiGetServiceName(QString const&) in libQt6Network.a(qauthenticator.cpp.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Note: The QT I used is built by myself. It works just fine If I don't use vcpkg.
And I noticed that in vcpkg port of qtbase GSSAPI is disabled for some reason, https://github.com/microsoft/vcpkg/blob/147027390bd5bb13ce14e0bfc153873e424b71b2/ports/qtbase/portfile.cmake#L132

Maybe there is something wrong with vcpkg's find GSSAPI?

**Environment**
- OS: MacOS 12.3
- Compiler: AppleClang 13.1.6.13160021 come with Xcode13 or CommandLineTools

**To Reproduce**
1. built QT[static] yourself by following [Building_Qt_6_from_Git](https://wiki.qt.io/Building_Qt_6_from_Git)
2. And use QtNetwork in your application
3. Try to build your application with CMAKE_TOOL_CHAIN configured to `vcpkg.cmake`.
Repro code when

**Expected behavior**
A success build

**Failure logs**
```
Undefined symbols for architecture arm64:
  "___gss_c_nt_hostbased_service_oid_desc", referenced from:
      qGSsapiGetServiceName(QString const&) in libQt6Network.a(qauthenticator.cpp.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Additional context**

**Comment by dg0yt** [comment]
Frameworks...

In the past problem there were problems when headers or libraries from frameworks were inconsistently found before those from vcpkg. In particular, this issue affected Github Actions workflows due to the pre-installed Mono Framework. A recent change fixed this by setting `CMAKE_FIND_FRAMEWORK` and `CMAKE_FIND_APPBUNDLE` to `LAST`: #23202. We think this is the right change for vcpkg.

Possible solutions for your context:
- Change those variables in your project.
- Directly initialize the `GSSAPI_INCLUDE_DIRS` and `GSSAPI_LIBRARIES` input variables (cf. `https://github.com/qt/qtbase/blob/dev/cmake/FindGSSAPI.cmake`).

**Comment by ylz-at** [comment]
@dg0yt I installed `qtbase` with vcpkg and it works. I don't use GSSAPI anyway.

**Comment by LilyWangLL** [comment]
cc @Neumann-A Could you help take a look this issue? Thanks in advance.

**Comment by Neumann-A** [comment]
> cc @Neumann-A Could you help take a look this issue? Thanks in advance.

I have no MacOSX dev platform. I am however wondering why the feature is not completely deactivated.

**Comment by aumuell** [comment]
This problem is not specific to vcpkg, but might occur whenever `CMAKE_FIND_FRAMEWORK=LAST` is set. See [QTBUG-114537](https://bugreports.qt.io/browse/QTBUG-114537).

**Comment by LilyWangLL** [comment]
This bug has been fixed in qt version 6.6, so I marked this issue as completed.
